nsxt/resource_nsxt_policy_nat_rule.go

```diff
@@ -284,8 +302,14 @@ func resourceNsxtPolicyNATRuleRead(d *schema.ResourceData, m interface{}) error
 		return handleMultitenancyTier0Error()
 	}
 
-	action := d.Get("action").(string)
-	natType := getNatTypeByAction(action)
+	natType := d.Get("type").(string)
+	if natType == "" {
+		// This can happen when provider was upgraded and we're refreshing an existing resource
+		// This is not an import case, so action should be set
+		action := d.Get("action").(string)
+		natType = getNatTypeByAction(natType, action)
+		d.Set("type", natType)
```

Comment:
Upon refresh, this will update the Terraform state with Type, and will avoid having a diff when doing the planning, is that correct?

Reply:
Correct